modules/openapi-generator/src/main/resources/php/HeaderSelector.mustache

```diff
@@ -73,7 +73,7 @@ class HeaderSelector
         if (count($accept) === 0 || (count($accept) === 1 && $accept[0] === '')) {
             return null;
         } elseif (preg_grep("/application\/json/i", $accept)) {
-            return 'application/json';
+            return $accept;
```

Comment:
This is incorrect: the method must return a string and you're returning a list of content types here.

Your intent seems to be returning all content types, which is what the fallback else-clause does. The question is: why look for application/json in the first place?

There is a relevant unit test for this specific case in `HeaderSelectorTest` and this is what we're breaking when we remove the `preg_grep` clause:

```
        $headers = $selector->selectHeaders([
            'application/xml',
            'application/json'
        ], []);
        $this->assertSame('application/json', $headers['Accept']);
```

The test case seems to make sure json content types are always used in favour of non-json types. If we want to keep that behaviour I would suggest something like:

```
        if (count($accept) === 0 || (count($accept) === 1 && $accept[0] === '')) {
            return null;
        } 

        $jsonAcceptTypes = preg_grep('#application/(ld\+)json#i');

        if (!empty($jsonAcceptTypes)) {
            $accept = $jsonAcceptTypes;
        }
        
        return implode(',', $accept);
```

This way the behaviour is unchanged if no json content type is passed to the method.

Reply:
got ya. I missed the single string return type. Funny this patch still works :) i guess in our case it's something like `[app/json+ld, app/json]` so all match JSON.

Maybe im missing the purpose of this method, as now looking at it.. it looks like it could be simply `implode(', ', $accept)` in all cases... but i'll do a more in-depth analysis tomorrow of what's exactly happening :+1: 

from your patch i figure we're filtering JSON-only accept types, while preserving the original input.